### DISCUSSION TITLE
Automatic subdivisions

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#715] [Q&A]
Hi All,
When EPOCH starts, the first thing we see on the screen it is doing is to divide simulation mesh into subdivisions in the x-y-z directions by one per processor core. I am curious is it possible to set these subdivisions manually ? Automatic subdivisions look often totally unreasonable and small change of initial settings might wildly change them. Also do these subdivisions change over the course of simulations?

**Comment by TomGoffrey** [comment]
Yes, you can set nproc{x,y,z} as explained  in the documentation: https://epochpic.github.io/documentation/input_deck/input_deck_control.html

The subdivisions may change depending on whether you have load balancing enabled.

**Comment by DanRRRR** [comment]
Many thanks Tom. It took my last hair to find it, including Search with keywords "load balancing",  "redistribution", "load", "balance" etc was not very specific, definitely needs AI to be more insightful.  The manual is  ~60MB of text, 10x of "War and Peace" size.  

And even after your hint we spent an hour trying to figure out how to setup it. Users should contribute a bit more and leave  more suggestions to improve the EPOCH. 

My would be that besides the large size the Manual is also a bit too cryptic. With just a few percents to its size more may substantially improve its usability adding small examples where possible. For example with respect to this redistribution problem the Manual says:

"nproc{x,y,z} - Number of processes in the x,y,z directions. By default, EPOCH will try to pick the best method of splitting the domain amongst the available processors but occasionally the user may wish to override this choice."

We tried: 

nproc= 4  4  8  -  failed
nproc= 4,  4,  8 -  failed
nproc{x,y,z}= 4  4  8 -  failed
nproc {4  4  8} -  failed
nproc= {4  4  8} -  failed 
etc

While changed text  should be :
"nproc{x,y,z} - Number of processes in the x,y,z directions (Example: nprocx=4, on a separate line nprocy=4, etc. ). By default, EPOCH will try to pick the best method of splitting the domain amongst the available processors but occasionally the user may wish to override this choice."

This is why I RTFM :)
Again  thanks for the help, sometimes even simple Yes or No in help responses reduce the time to find the solution immensely

**Reply by Status-Mirror** [reply]
I've added a quick example to that entry on the website.
